content/api-streamer/producer.textile

```diff
@@ -0,0 +1,239 @@
+
+<!--TODO check code examples for accuracy-->
+
+The following example shows a mock data stream by repeating the publish action at regular intervals.
+
+```[javascript]
+  var rest = new Ably.Rest('{{API_KEY}}');
+  var channel = rest.channels.get('{{RANDOM_CHANNEL_NAME}}');
+  setInterval(function(){ 
+      channel.publish('example', 'message data'); 
+  }, 3000);
+```
+
+```[nodejs]
+  var rest = new Ably.Rest('{{API_KEY}}');
+  var channel = rest.channels.get('{{RANDOM_CHANNEL_NAME}}');
+  setInterval(function(){ 
+      channel.publish('example', 'message data'); 
+  }, 3000);
+```
+
+```[ruby]
+  rest = Ably::Rest.new('{{API_KEY}}')
+  channel = rest.channels.get('{{RANDOM_CHANNEL_NAME}}')
+  while true
+    channel.publish 'example', 'message data'
+    sleep 3
+  end
+```
+
+```[python]
+  rest = AblyRest('{{API_KEY}}')
+  channel = rest.channels.get('{{RANDOM_CHANNEL_NAME}}')
+  def setInterval(func,time):
+    e = threading.Event()
+    while not e.wait(time):
+        func()
+
+  def publish():
+    channel.publish(u'example', u'message data')"
+
+  setInterval(publish,3)
+```
+
+```[php]
+  $rest = new Ably\AblyRest('{{API_KEY}}');
+  $channel = $rest->channels->get('{{RANDOM_CHANNEL_NAME}}');
+  function setInterval($f, $milliseconds){
+    $seconds=(int)$milliseconds/1000;
+    while(true){
+        $f();
+        sleep($seconds);
+    }
+  }
+  setInterval(function() use($channel) {
+   $channel->publish('example', 'message data');
+  }, 3000);
+```
+
+```[java]
+  AblyRest rest = new AblyRest("{{API_KEY}}");
+  Channel channel = rest.channels.get("{{RANDOM_CHANNEL_NAME}}");
+  new Timer().scheduleAtFixedRate(new TimerTask(){
+    @Override
+    public void run(){
+       channel.publish("example", "message data");
+    }
+  },0,3000);
+```
+
+```[csharp]
+  AblyRest rest = new AblyRest("{{API_KEY}}");
+  var channel = rest.Channels.Get("{{RANDOM_CHANNEL_NAME}}");
+  SetInterval(() => channel.PublishAsync("example", "message data"), TimeSpan.FromSeconds(3));
+```
+
+```[objc]
+  ARTRest *rest = [[ARTRest alloc] initWithKey:@"{{API_KEY}}"];
+  ARTRestChannel *channel = [rest.channels get:@"{{RANDOM_CHANNEL_NAME}}"];
+  while True:
+    [channel publish:@"example" data:@"message data"];
+    time.sleep(3)
+```
+
+```[swift]
+  let rest = ARTRest(key: "{{API_KEY}}")
+  let channel = rest.channels.get("{{RANDOM_CHANNEL_NAME}}")
+  var publishTimer = Timer.scheduledTimer(timeInterval: 3.0, target: self, selector: #selector(ViewController.publishData), userInfo: nil, repeats: true)
+  @objc func publishData(){
+    channel.publish("example", data: "message data")
+  }
+```
+
+```[go]
+  rest, err := ably.NewRestClient(ably.NewClientOptions("{{API_KEY}}"))
+  channel := rest.Channels.Get("{{RANDOM_CHANNEL_NAME}}", nil)
+  interval := setInterval(func() {
+            err = channel.Publish("example", "message data")
+  }, 3000, false)
+  
+```
+
+h4(#deploy-using-realtime-sdks). Deploying data streams using Ably's Realtime Client SDKs
```

Comment:
Not sure if having REST + Realtime example here is necessary, given both look exactly the same besides for the most part.